support-frontend/assets/components/directDebit/directDebitProgressiveDisclosure/directDebitForm.jsx

```diff
@@ -0,0 +1,289 @@
+      // Disabling this as react docs say it is okay to set state here with conditions
+      /* eslint-disable react/no-did-update-set-state */
+      this.setState({
+        allErrorsLength: this.props.allErrors.length,
+      });
+    }
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (nextProps.allErrors.length !== prevState.allErrorsLength) {
+      return { allErrorsLength: nextProps.allErrors.length };
+    }
+    return null;
+  }
+
+  onSubmit = (event) => {
+    event.preventDefault();
+    this.props.validateForm();
+    this.handleErrorsAndCheckAccount();
+  }
+
+  onChange = (field, dispatchUpdate, event) => {
+    this.setState({
+      [field]: {
+        ...this.state[field],
+        error: '',
+      },
+    });
+    dispatchUpdate(event);
+  }
+
+  getAccountErrors = (state) => {
+    const cardErrors = fieldNames.map(field =>
+      ({ message: state[field].error }));
+    return cardErrors;
+  }
+
+  getAccountErrorsLength = (cardErrors) => {
+    let accum = 0;
+    cardErrors.forEach((item) => {
+      if ((item.message).length > 0) {
+        accum += 1;
+      }
+    });
+    return accum;
+  }
+
+  handleErrorsAndCheckAccount = () => {
+    const { props } = this;
+    let accountErrorsLength = 0;
+    fieldNames.forEach((field) => {
+      // The following line is checking that the field value matches the validation rule
+      if (!this.state[field].rule(props[field])) {
+        // If not, an error is set in state
+        this.setState(
+          state => ({
+            [field]: {
+              ...state[field],
+              error: this.state[field].message,
+            },
+          }),
+          // And then the error count in state is updated
+          () => {
+            accountErrorsLength = this.getAccountErrorsLength(this.getAccountErrors(this.state));
+            this.setState({
+              accountErrorsLength,
+            });
+          },
+        );
+      } else {
+        // If the field is fine, the number of errors is updated
+        this.setState(
+          state => ({
+            accountErrorsLength: this.getAccountErrorsLength(this.getAccountErrors(state)),
+          }),
+          // And then all the error count is checked before an action is dispatched to check the account
+          () => {
+            if (this.state.allErrorsLength === 0 && this.state.accountErrorsLength === 0) {
+              props.payDirectDebitClicked();
+            }
+          },
+        );
+      }
+    });
+  }
+
+  submitForm = () => {
+    const { props } = this;
+    props.confirmDirectDebitClicked(props.onPaymentAuthorisation);
+    props.submitForm();
+  }
+
+  render() {
+    const { props, state } = this;
+    const accountErrors = this.getAccountErrors(state);
+    const { accountErrorsLength } = state;
+    const showGeneralError = props.allErrors.length === 0 && accountErrorsLength === 0 &&
+    ((props.submissionError !== null && props.submissionError !== null) ||
```

Comment:
You're checking `props.submissionError` twice here

Reply:
Oops!